providers/openmetrics/client_metrics.go

```diff
@@ -123,65 +127,13 @@ func (m *ClientMetrics) Collect(ch chan<- openmetrics.Metric) {
 	m.clientHandledCounter.Collect(ch)
 	m.clientStreamMsgReceived.Collect(ch)
 	m.clientStreamMsgSent.Collect(ch)
-	if m.clientHandledHistogramEnabled {
+	if m.clientHandledHistogram != nil {
 		m.clientHandledHistogram.Collect(ch)
 	}
-	if m.clientStreamRecvHistogramEnabled {
+	if m.clientStreamRecvHistogram != nil {
 		m.clientStreamRecvHistogram.Collect(ch)
 	}
-	if m.clientStreamSendHistogramEnabled {
+	if m.clientStreamSendHistogram != nil {
 		m.clientStreamSendHistogram.Collect(ch)
 	}
 }
-
```

Comment:
All of that has been turned into options for the constructor. `ClientMetrics` is now immutable (from the outside) and there is no chance of a data race because someone calls these methods.